assets/js/panel.js

```diff
@@ -1,28 +1,86 @@
 import { Leaf } from "./leaves";
 
 class Panel {
+  /*
+    The panel is used to display the project introduction on page load
+    and leaf details as leaves are selected.
+    On desktop, leaf details can be closed but the panel is always visible.
+    On mobile, the panel can be closed completely; there is an info
+    button to redisplay the project introduction.
+
+    The class `closed`` is used to indicate the panel is closed
+    (hidden on mobile, display introduction on desktop); the class
+    `show-details` is used to indicate whether leaf details are visible or not.
+
+    */
+
   constructor() {
     // get a reference to the panel element
     this.el = document.querySelector("#leaf-details");
+    this.infoButton = document.querySelector("header .info");
     this.bindHandlers();
   }
 
-  close() {
-    // close the panel
-    this.el.parentElement.classList.remove("show-details");
-    this.el.parentElement.classList.add("closed");
+  open(showDetails = true) {
+    // open the panel; show leaf details by default
+    // disable the info button
+    let container = this.el.parentElement;
+    if (showDetails) {
+      container.classList.add("show-details");
+    } else {
+      // when show details is not true, ensure leaf details are hidden
+      container.classList.remove("show-details");
+    }
+    container.classList.remove("closed");
+    // disable the info button; inactive when the intro is visible
+    this.infoButton.disabled = true;
+  }
+
+  close(closeDetails = true) {
+    // close the intro and enable the info button
+    // by default, closes panel entireuly
+    let container = this.el.parentElement;
+    // determine if leaf details are currently displayed
+    let leafVisible = container.classList.contains("show-details");
+
+    // if leaf details are visible and close details is true,
+    // deselct the leaf currently displayed, close that also,
+    // unless closeDetails has been disabled;
+    // (has a side effect of also removing any currently selected tag)
+    if (leafVisible && closeDetails) {
+      container.classList.remove("show-details");
+      Leaf.closeLeafDetails();
+    }
+
+    // if we are closing everything or no leaf is visible, close the panel
+    if (closeDetails || !leafVisible) {
+      container.classList.add("closed");
+    }
+
+    // enable the info button;
+    // provides a way to get back to the intro on mobile
+    this.infoButton.disabled = false;
+  }
+
+  showIntro() {
+    // showing the intro implies closing leaf details (if a leaf is selected)
     Leaf.closeLeafDetails();
+    // open the panel without showing leaf details section
+    this.open(false);
+  }
+
+  closeIntro() {
+    // close the panel without closing leaf details
+    this.close(false);
   }
```

Comment:
ooh nice. I'd be tempted to add a `showError()` function here for my case (https://github.com/Princeton-CDH/lenape-timetree/issues/132) as well

Reply:
yeah, but the problem is that the leaf option doesn't know about / have access to the panel object 

Maybe a show leaf error on the leaf object, though?